Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -981,16 +981,26 @@ void SourceBuffer::evictCodedFrames(size_t newDataSize)
     if (isRemoved())
         return;
 
+    const auto hasTooManySamples = [&] {
+        const size_t evictionThreshold = m_private->platformEvictionThreshold();
+        if (!evictionThreshold)
+            return false;
+        size_t currentSize = 0;
+        for (const auto& trackBuffer : m_trackBufferMap.values())
+            currentSize += trackBuffer.samples.size();
+        return currentSize > evictionThreshold;
+    };
+
     // This algorithm is run to free up space in this source buffer when new data is appended.
     // 1. Let new data equal the data that is about to be appended to this SourceBuffer.
     // 2. If the buffer full flag equals false, then abort these steps.
-    if (!m_bufferFull)
+    if (!m_bufferFull && !hasTooManySamples())
```

Comment:
Exactly, if the threshold is not set, that function evaluates to false and this condition will be true, so we will be skipping eviction. That can't happen. In non GStreamer ports eviction would not work (when the buffer is not full) because the threshold is always 0, `hasTooManySamples` evaluates to `false`, this condition would evaluate to `true` and there you are, eviction skipped. Other than screwing other ports, you also screw manual eviction by the JS app.